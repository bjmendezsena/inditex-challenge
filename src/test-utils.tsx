import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { theme } from "./theme";

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: process.env.NODE_ENV === "test" ? () => {} : console.error,
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const DefaultTheme = extendTheme(theme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
