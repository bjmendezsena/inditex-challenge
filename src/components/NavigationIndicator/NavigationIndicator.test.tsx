import { screen, waitFor } from "@testing-library/react";
import { render } from "../../test-utils";
import {
  NavigationIndicator,
  NavigationIndicatorProps,
} from "./NavigationIndicator";

describe(`<${NavigationIndicator.name}/> component`, () => {
  const factoryComponent = (props?: NavigationIndicatorProps) =>
    render(<NavigationIndicator {...props} />);

  test("renders correctly", () => {
    const view = factoryComponent();
    expect(view.asFragment()).toMatchSnapshot();
  });

  test("No display when isNavigating is false", async () => {
    factoryComponent();
    const container = screen.getByTestId("navigation-indicator");

    await waitFor(() => {
      expect(container).toHaveStyle("display: none");
    });
  });
  test("show indicator when isNavigating is true", async () => {
    factoryComponent({
      isNavigating: true,
    });
    const container = screen.getByTestId("navigation-indicator");

    await waitFor(() => {
      expect(container).toHaveStyle("display: block");
    });
  });
});
