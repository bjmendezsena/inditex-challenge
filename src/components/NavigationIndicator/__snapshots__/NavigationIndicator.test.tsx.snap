// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationIndicator/> component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-15vddy2"
    data-testid="navigation-indicator"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
